TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/NettyPacketToHttpConsumer.java

```diff
@@ -66,29 +68,39 @@ public class NettyPacketToHttpConsumer implements IPacketFinalizingConsumer<Aggr
     private Channel channel;
     AggregatedRawResponse.Builder responseBuilder;
     IWithTypedEnclosingScope<IReplayContexts.ITargetRequestContext> currentRequestContextUnion;
+    Duration readTimeoutDuration;
 
-    private static class ConnectionClosedListenerHandler extends ReadTimeoutHandler {
+    private static class ConnectionClosedListenerHandler extends ChannelInboundHandlerAdapter {
         private final IReplayContexts.ISocketContext socketContext;
-        ConnectionClosedListenerHandler(IReplayContexts.IChannelKeyContext channelKeyContext,
-                                        Duration timeout) {
-            super(timeout.toMillis(), TimeUnit.MILLISECONDS);
+        ConnectionClosedListenerHandler(IReplayContexts.IChannelKeyContext channelKeyContext) {
             socketContext = channelKeyContext.createSocketContext();
         }
         @Override
         public void channelUnregistered(ChannelHandlerContext ctx) throws Exception {
             socketContext.close();
             super.channelUnregistered(ctx);
         }
+
+        @Override
+        public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
+            socketContext.close();
```

Comment:
You should add a line beforehand for `socketContext.addTraceException(cause, true)`

Reply:
Was the socketContext NOT being closed when there was an exception?  Does an exception skip unregistration?
I think it's safe to call close() twice, but can you check?  If it isn't, we should have a check in this class (like nulling out the socketContext)

Reply:
You're right. Socket context would have been closed twice with this in cases an exception reaches ConnectionClosedListenerHandler. It doesn't occur in the timeout case since that handler is after this one.

It doesn't look like .close should be called multiple times, it seems that would double count metrics. I'll do a couple changes to fix this.
1. remove the  socketContext.close from exception caught
2. change channelUnregistered -> channelInactive. Technically, unregistered can be called multiple times, but inactive is only called once on channel close

Reply:
In our use, channel is only unregistered when it's closed, but there can be implementations where channel is unregistered from the event loop and re-registered again so inactive follows the contracts better